File: 0118. Pascal's Triangle/Solution0118.cpp
#include <iostream>
#include <string>
#include<vector>
using namespace std;

class Solution0118 {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> res;
        for (int i = 0; i < numRows; i++) {
            vector<int> row(i+1, 1);
            for (int j = 1; j < i; j++) {
                row[j] = res[i-1][j-1] + res[i-1][j];
            }
            res.push_back(row);
        }
        return res;
    }
};

int main() {
    Solution0118 sol;
    int numRows = 5;
    vector<vector<int>> vec = sol.generate(numRows);
    for (int i = 0; i < vec.size(); ++i)
    {
        for (int j = 0; j < vec[i].size(); ++j)
        {
            cout << vec[i][j];
        }
        cout << std::endl;
    }
    
}

File: 0118. Pascal's Triangle/Solution0118.java
import java.util.*;

class Solution0118 {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<List<Integer>>();
        if (numRows == 0) {
            return res;
        }
        List<Integer> prev = new ArrayList<Integer>();
        List<Integer> cur = new ArrayList<Integer>();
        cur.add(1);
        res.add(cur);
        prev = cur;
        for (int i = 1; i < numRows; i++) {
            cur = new ArrayList<Integer>();
            cur.add(1);
            for (int j = 1; j < prev.size(); j++) {
                cur.add(prev.get(j) + prev.get(j-1));
            }
            cur.add(1);
            res.add(cur);
            prev = cur;
        }
        return res;
    }

    public static void main(String[] args) {
        Solution0118 sol = new Solution0118();
        int numRows = 5;
        List<List<Integer>> res = sol.generate(numRows);
        System.out.println(Arrays.deepToString(res.toArray()));
    }
}